Define the WebContent process as launched in WebPage::create

pvollan opened the pull request:
#### b21d74bab849d7cc98e7007ef107b3caf445cfe6
<pre>
Define the WebContent process as launched in WebPage::create
<a href="https://bugs.webkit.org/show_bug.cgi?id=257964">https://bugs.webkit.org/show_bug.cgi?id=257964</a>
rdar://110648262

Reviewed by Brent Fulgham.

From a sandbox perspective, define the WebContent process as launched in WebPage::create. This ensures that the
sandbox state variable is set before any content is loaded, which is not immediately clear from doing it when
the load is committed.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::create):
(WebKit::WebPage::didCommitLoad):

Canonical link: <a href="https://commits.webkit.org/265080@main">https://commits.webkit.org/265080@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1232214de7054ad4057bafbd71fb4620c6e333c4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9733 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9984 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10229 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11392 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9498 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11969 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9942 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12421 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9887 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10721 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8278 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11552 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8025 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16247 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9124 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8990 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12325 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9489 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8665 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2342 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12891 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9259 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -483,6 +483,11 @@ Ref<WebPage> WebPage::create(PageIdentifier pageID, WebPageCreationParameters&&
 
+#if HAVE(SANDBOX_STATE_FLAGS)
+    auto auditToken = WebProcess::singleton().auditTokenForSelf();
+    sandbox_enable_state_flag("WebContentProcessLaunched", *auditToken);
```
Is the intention of this change to always be in a "launched" state by the time we begin processing page content? I think that makes sense, as the process has to finish launching to be ready to begin handling things.

pvollan replied:
Yes, that is correct. The sandbox has quite a few rules that depend on this variable.

Thanks for reviewing!

szewai replied:
IIUC web process could create more than one web pages. It does not seem like this flag could change. Should we ensure the flag is set only once for efficiency?

cdumez replied:
Sihui is right that we can create more than once WebPage in a WebProcess and thus call this several times. However, note that the previous code was calling it in didCommitLoad() and it is extremely common to commit more than one load in single WebProcess as well. Therefore, this is not a "regression".

That said, if the call is expensive, it is still a good idea to only call once, maybe with a global static boolean.

pvollan replied:
That is a very good point. I can make a patch for that.

Thanks for reviewing!

brentfulgham reviewed (approved):
I think this change seems reasonable. r=me.

pvollan committed:
Define the WebContent process as launched in WebPage::create
https://bugs.webkit.org/show_bug.cgi?id=257964
rdar://110648262

Reviewed by Brent Fulgham.

From a sandbox perspective, define the WebContent process as launched in WebPage::create. This ensures that the
sandbox state variable is set before any content is loaded, which is not immediately clear from doing it when
the load is committed.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::create):
(WebKit::WebPage::didCommitLoad):

Canonical link: https://commits.webkit.org/265080@main
